### ISSUE TITLE
maximum recursion depth exceeded

### ISSUE THREAD

**Issue by altaurog** [PyCQA/astroid#542]
I realize there are a bunch of issues like this, but here’s another example.  Apologies if it doesn’t add anything of value.

### Steps to reproduce
1. analyse the following code:

```python
"""GraphQL Schema"""
import graphene
import graphene.relay as relay


sites = [
    'Facebook',
    'Github',
    'Instagram',
    'LinkedIn',
    'Pinterest',
    'Reddit',
    'Slack',
    'Snapchat',
    'Twitter',
]


def resolve_sites(*_, **__):
    """retrieve multiple sites--ignore args for simplicity’s sake"""
    res = [Site(id=i, name=n) for i, n in enumerate(sites)]
    edges = make_edges(res)
    page_info = graphene.PageInfo(
        has_previous_page=False,
        has_next_page=False,
        start_cursor=res[0].id,
        end_cursor=res[-1].id,
    )
    return SiteConnection(page_info=page_info, edges=edges, totalCount=len(sites))


def make_edges(objs):
    return [SiteConnection.Edge(node=s, cursor=s.id) for s in objs]


class Site(graphene.ObjectType):
    id = graphene.String(required=True)
    name = graphene.String(required=True)


class SiteConnection(graphene.Connection):
    totalCount = graphene.Int(required=True)
    class Meta:
        node = Site


class Query(graphene.ObjectType):
    sites = graphene.ConnectionField(SiteConnection, resolver=resolve_sites)


schema = graphene.Schema(query=Query)
```

### Current behavior
```
...
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/context.py", line 71, in cache_generator
    for result in generator:
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/decorators.py", line 86, in wrapped
    generator = _func(node, context, **kwargs)
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/inference.py", line 138, in infer_name
    frame, stmts = self.lookup(self.name)
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/node_classes.py", line 989, in lookup
    return self.scope().scope_lookup(self, name)
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/scoped_nodes.py", line 2012, in scope_lookup
    return frame._scope_lookup(node, name, offset)
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/scoped_nodes.py", line 160, in _scope_lookup
    stmts = node._filter_stmts(self.locals[name], self, offset)
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/node_classes.py", line 1030, in _filter_stmts
    myframe = self.frame().parent.frame()
  File "/home/altaurog/.venv/org_graph_api/lib/python3.6/site-packages/astroid/node_classes.py", line 428, in frame
    return self.parent.frame()
RecursionError: maximum recursion depth exceeded
```

### Expected behavior

No recursion exception.

### Workaround:

With the following change to the code under analysis, it does not fail:
```diff
22c22
<     edges = make_edges(res)
---
>     edges = make_edges(SiteConnection.Edge, res)
32,33c32,33
< def make_edges(objs):
<     return [SiteConnection.Edge(node=s, cursor=s.id) for s in objs]
---
> def make_edges(edge_type, objs):
>     return [edge_type(node=s, cursor=s.id) for s in objs]
```

### version info
```
$ python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"
1.6.2
```

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by altaurog** [comment]
No, thank you!

**Comment by jkonecny12** [comment]
We are experiencing the same issue:

File: 
https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/bootloader.py

Version:
`2.0.0.dev`

Output:
https://jkonecny.fedorapeople.org/others/astroid-bug.txt


I'm not sure what is causing this inside of that file. Is there a way how to find the problematic line?

**Comment by brycepg** [comment]
This issue was fixed by 180aec84a2699158c52572cced82d79b443203cc !
